### ISSUE TITLE
v1.2 breaks "pressed" and "released" functionality

### ISSUE THREAD

**Issue by pevzi** [tesselode/tactile#4]
The update that was meant to make calls to `Button:update()` unnecessary actually breaks the `pressed()` and `released()` functionality in some cases.
Say, we've released a key in the current frame and there is an `isDown()` check. It means that `update()` gets called which sets the `downPrev` field to `true` and `down` to `false`. We get into a situation where we're unable to "catch" that release event anymore because the second call to `update()` (which would be performed by `released()`) would set `downPrev` to `false`.
In other words, `update()` shouldn't be called more than once in a single frame and the current API makes it possible to do it unintentionally.
This issue actually affects my game prototype and breaks the jumping functionality ([here](https://github.com/pevzi/yummy-octo-duck/blob/master/entities/player.lua#L65)'s the relevant code). Probably this code doesn't represent the most straightforward way to implement the features I needed but I still think that the behavior described above isn't the one that is expected here.

**Comment by shakesoda** [comment]
I was concerned about this when I saw the update, but wasn't entirely sure if love's isDown functions would actually change state between calls to love.event.poll().

**Comment by tesselode** [comment]
Shoot. I'm really stupid. Can you like, delete commits and roll back to a previous release? What's standard procedure for taking back a bad version release?

**Comment by pevzi** [comment]
You can use `git revert fad81df -m 1` which will create a commit that undoes the changes introduced by the merge commit (and since it is a merge commit we also need to specify the parent with `-m`). Or you can `git reset --hard d3b2f3a` and then `git push -f` but this is generally discouraged.

**Comment by shakesoda** [comment]
Yeah, don't push -f a hard reset, that'll make git refuse to pull for existing checkouts.

**Comment by tesselode** [comment]
Alright, we should be back to v1.1.1. This is what I get for releasing things hastily and not having a proper test suite. :\

Thanks for the PR, though, that was a good catch!
